jwql/website/apps/jwql/monitor_pages/monitor_readnoise_bokeh.py

```diff
@@ -12,42 +12,55 @@
 
     ::
 
-        from jwql.website.apps.jwql import monitor_pages
+        .
         monitor_template = monitor_pages.ReadnoiseMonitor()
         monitor_template.input_parameters = ('NIRCam', 'NRCA1_FULL')
 """
 
 from datetime import datetime, timedelta
 import os
 
+from bokeh.embed import components
+from bokeh.layouts import column, row
+from bokeh.models import Panel, Tabs  # bokeh <= 3.0
+from bokeh.models import ColumnDataSource, HoverTool
+# from bokeh.models import TabPanel, Tabs  # bokeh >= 3.0
+from bokeh.plotting import figure
 import numpy as np
 
-from jwql.bokeh_templating import BokehTemplate
 from jwql.database.database_interface import session
 from jwql.database.database_interface import FGSReadnoiseStats, MIRIReadnoiseStats, NIRCamReadnoiseStats, NIRISSReadnoiseStats, NIRSpecReadnoiseStats
-from jwql.utils.constants import JWST_INSTRUMENT_NAMES_MIXEDCASE
+from jwql.utils.constants import FULL_FRAME_APERTURES, JWST_INSTRUMENT_NAMES_MIXEDCASE
 
-SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
 
+class ReadnoiseMonitorData():
+    """Class to hold bias data to be plotted
 
-class ReadnoiseMonitor(BokehTemplate):
+    Parameters
+    ----------
 
-    # Combine the input parameters into a single property because we
-    # do not want to invoke the setter unless all are updated
-    @property
-    def input_parameters(self):
-        return (self._instrument, self._aperture)
+    instrument : str
+        Instrument name (e.g. nircam)
 
-    @input_parameters.setter
-    def input_parameters(self, info):
-        self._instrument, self._aperture = info
-        self.pre_init()
-        self.post_init()
+    Attributes
+    ----------
+
+    instrument : str
+        Instrument name (e.g. nircam)
+    aperture : str
+        Aperture name (e.g. apername)
```

Comment:
Should this include "tabs", "plots", "query_results" etc?

Reply:
I think that is valid for the classes where they are generated. I think that `stats_table` and `query_results` should be added here for sure. I will make sure to do that!